src/sexpr.rs

```diff
@@ -81,10 +98,10 @@ struct ArenaInner {
     id: ArenaId,
 
     /// Interned strings.
-    atoms: Vec<String>,
+    strings: Vec<String>,
 
-    /// Backwards lookup for string data.
-    atom_map: HashMap<&'static str, SExpr>,
+    /// Backwards lookup for interned strings.
+    string_map: HashMap<&'static str, u32>,
```

Comment:
This is an existing issue, but we might *technically* need to wrap `string_map` in `ManuallyDrop` and and have a custom `impl Drop` that ensures that `string_map` is dropped before `strings` so that we don't have any dangling references to deallocated strings, which I am pretty sure is technically UB even if we never touch the dangling references.

Reply:
Could be good to run some tests/examples under MIRI in CI to catch this kind of thing.

Reply:
Let's do that in a follow-up.